apps/src/lib/levelbuilder/data-docs-editor/DataDocEditAll.jsx

```diff
@@ -1,11 +1,97 @@
-import React from 'react';
+import React, {useState} from 'react';
+import $ from 'jquery';
+import PropTypes from 'prop-types';
+import {Link, TextLink} from '@dsco_/link';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import Dialog, {Title as DialogTitle} from '@cdo/apps/templates/Dialog';
 
 const DataDocEditAll = props => {
+  const {dataDocs: initialDataDocs} = props;
+  const [dataDocs, setDataDocs] = useState(initialDataDocs);
+  const [showDeleteWarningDialog, setShowDeleteWarningDialog] = useState(false);
+  const [pendingDeleteDocKey, setPendingDeleteDocKey] = useState(null);
+
+  const initiateDeleteDataDoc = docKeyToDelete => {
+    setPendingDeleteDocKey(docKeyToDelete);
+    setShowDeleteWarningDialog(true);
+  };
+
+  const deleteDataDoc = () => {
+    $.ajax({
+      url: `/data_docs/${pendingDeleteDocKey}`,
+      method: 'DELETE'
+    }).done(() => {
+      setDataDocs([
+        ...dataDocs.filter(
+          dataDoc => !pendingDeleteDocKey.includes(dataDoc.key)
+        )
+      ]);
+      setShowDeleteWarningDialog(false);
+      setPendingDeleteDocKey(null);
+    });
+  };
```

Comment:
This is a bit of an edge case, but I am seeing a problem if I open the edit page in two places, and try to delete the same object twice. I would suggest fixing this while we are thinking about it.

when I try to delete the nonexistent item, I get a 404, which is fine, but the dialog doesn't close, and if you close it manually, the item is still in the list: 
<img width="1540" alt="Screen Shot 2022-10-07 at 3 07 39 PM" src="https://user-images.githubusercontent.com/8001765/194670123-69c03072-104d-4ca2-821d-fdf4ae5b8104.png">

What I would suggest is, `catch` here, and if the status code was 404, just close the dialog and remove the item from the list like you are doing in `done` above. you _could_ show an error saying the object was already deleted... but... I'm not sure how helpful that is?

Reply:
thank you for catching that, good call!

I wonder if the server isn't responding then the dialog not closing automatically would visually indicate something went wrong moreso than the dialog closing without deleting anything?

Reply:
Yep, agreed!

Reply:
great - glad you are both on board with the proposed 404 behavior.

yes, in case of any error besides 404, it seems fine to leave the dialog open. levelbuilders are usually savvy enough to grab any details from the js error console and send those to us.